tests/OTIOClipTests.c

```diff
@@ -78,96 +72,71 @@ static void otio_clip_constructor_test(void **state) {
             Item_trimmed_range((Item *) clip, errorStatus);
-    RationalTime *clip_trimmed_range_duration =
+    RationalTime clip_trimmed_range_duration =
             TimeRange_duration(clip_trimmed_range);
 
     assert_true(RationalTime_equal(clip_duration, clip_trimmed_range_duration));
 
-    RationalTime *tr_duration = TimeRange_duration(tr);
+    RationalTime tr_duration = TimeRange_duration(tr.value);
 
     assert_true(RationalTime_equal(clip_duration, tr_duration));
 
-    assert_true(TimeRange_equal(tr, clip_trimmed_range));
+    assert_true(OptionalTimeRange_valid(tr));
+    assert_true(TimeRange_equal(OptionalTimeRange_value(tr),
+                                clip_trimmed_range));
 
-    TimeRange *clip_available_range = Clip_available_range(clip, errorStatus);
+    TimeRange clip_available_range = Clip_available_range(clip, errorStatus);
 
-    assert_true(TimeRange_equal(tr, clip_available_range));
-
-    RationalTime_destroy(start);
-    start = NULL;
-    RationalTime_destroy(duration);
-    duration = NULL;
-    RationalTime_destroy(clip_duration);
-    clip_duration = NULL;
-    TimeRange_destroy(clip_trimmed_range);
-    clip_trimmed_range = NULL;
-    RationalTime_destroy(clip_trimmed_range_duration);
-    clip_trimmed_range_duration = NULL;
-    TimeRange_destroy(clip_available_range);
-    clip_available_range = NULL;
+    assert_true(TimeRange_equal(OptionalTimeRange_value(tr),
+                                clip_available_range));
 
     start = RationalTime_create(86500, 24);
     duration = RationalTime_create(50, 24);
-    TimeRange *source_range =
-            TimeRange_create_with_start_time_and_duration(start, duration);
+    OptionalTimeRange source_range = OptionalTimeRange_create(
+            TimeRange_create_with_start_time_and_duration(start, duration));
     Item_set_source_range((Item *) clip, source_range);
     clip_duration = Item_duration((Item *) clip, errorStatus);
 
     assert_false(RationalTime_equal(clip_duration, tr_duration));
 
     clip_trimmed_range = Item_trimmed_range((Item *) clip, errorStatus);
 
-    assert_false(TimeRange_equal(clip_trimmed_range, tr));
+    assert_false(TimeRange_equal(clip_trimmed_range,
+                                 OptionalTimeRange_value(tr)));
 
-    TimeRange *clip_source_range = Item_source_range((Item *) clip);
-    RationalTime *clip_source_range_duration =
-            TimeRange_duration(clip_source_range);
+    OptionalTimeRange clip_source_range = Item_source_range((Item *) clip);
+    assert_true(OptionalTimeRange_valid(clip_source_range));
+    RationalTime clip_source_range_duration =
+            TimeRange_duration(OptionalTimeRange_value(clip_source_range));
 
     assert_true(RationalTime_equal(clip_source_range_duration, clip_duration));
 
-    assert_true(TimeRange_equal(clip_trimmed_range, clip_source_range));
+    assert_true(TimeRange_equal(clip_trimmed_range, OptionalTimeRange_value(clip_source_range)));
 
-    TimeRange_destroy(tr);
```

Comment:
And this chunk shows the payoff of all the work :)

Reply:
:relieved: